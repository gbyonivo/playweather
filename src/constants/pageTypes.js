export default {
  forecast: 'forecast',
  current: 'current'
};