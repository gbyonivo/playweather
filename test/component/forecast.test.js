import React from 'react';
import { createStore } from 'redux';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Forecast from '../../src/components/forecast';
import combinedReducer from '../../src/reducers';
import pageTypes from '../../src/constants/pageTypes';
import { forecast } from '../__testData__';

const props = {
  fetchData: () => { }
};

const state = {
  errorFetchingCurrentWeather: undefined,
  errorFetchingForecast: undefined,
  isFetchingCurrentWeather: false,
  isFetchingForecast: false,
  activeLocation: 'venice',
  forecast: {}
};

describe('Forecast component', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: state });
    const actual = mount(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey', match: { params: { type: pageTypes.current } } }]}>
        <Provider store={store}>
          <Forecast {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(actual).toMatchSnapshot();
  });
});

describe('Forecast component', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), { weatherReducer: { ...state, forecast } });
    const actual = mount(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey', match: { params: { type: pageTypes.current } } }]}>
        <Provider store={store}>
          <Forecast {...props} />
        </Provider>
      </MemoryRouter>
    );
    expect(actual).toMatchSnapshot();
  });
});