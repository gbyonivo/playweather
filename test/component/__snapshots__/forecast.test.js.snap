// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast component renders correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "match": Object {
          "params": Object {
            "type": "current",
          },
        },
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "match": Object {
              "params": Object {
                "type": "current",
              },
            },
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "match": Object {
            "params": Object {
              "type": "current",
            },
          },
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(Forecast)
        fetchData={[Function]}
      >
        <Forecast
          activeLocation="venice"
          dispatch={[Function]}
          fetchData={[Function]}
          forecast={Object {}}
          isFetchingForecast={false}
        >
          <div
            className="forecast"
          >
            <div
              className="message"
            >
              Type City and Country Code or just City Then Hit Search e.g. Manchester or Manchester, UK
            </div>
          </div>
        </Forecast>
      </Connect(Forecast)>
    </Provider>
  </Router>
</MemoryRouter>
`;

exports[`Forecast component renders correctly 2`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "match": Object {
          "params": Object {
            "type": "current",
          },
        },
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "match": Object {
              "params": Object {
                "type": "current",
              },
            },
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "match": Object {
            "params": Object {
              "type": "current",
            },
          },
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Connect(Forecast)
        fetchData={[Function]}
      >
        <Forecast
          activeLocation="venice"
          dispatch={[Function]}
          fetchData={[Function]}
          forecast={
            Object {
              "cod": 2,
              "list": Array [
                Object {
                  "dt_txt": "txt",
                  "main": Object {
                    "humidity": 10,
                    "temp_max": 11,
                    "temp_min": 9,
                  },
                  "weather": Array [
                    Object {
                      "icon": "10d",
                      "main": "terav",
                    },
                  ],
                },
              ],
            }
          }
          isFetchingForecast={false}
        >
          <div
            className="forecast"
          >
            <div>
              <h4>
                venice
              </h4>
              <div
                className="forecastList"
              >
                <ForecastListItem
                  item={
                    Object {
                      "dt_txt": "txt",
                      "main": Object {
                        "humidity": 10,
                        "temp_max": 11,
                        "temp_min": 9,
                      },
                      "weather": Array [
                        Object {
                          "icon": "10d",
                          "main": "terav",
                        },
                      ],
                    }
                  }
                >
                  <div
                    className="forecastListItem"
                  >
                    <h4>
                      txt
                    </h4>
                    <div
                      className="weatherIcons"
                    >
                      <WeatherIcons
                        weather={
                          Array [
                            Object {
                              "icon": "10d",
                              "main": "terav",
                            },
                          ]
                        }
                      >
                        <div
                          className="weatherIcons"
                        >
                          <div
                            className="weatherIcon"
                            key="10d0"
                          >
                            <img
                              alt=""
                              src="http://openweathermap.org/img/w/10d.png"
                            />
                          </div>
                        </div>
                      </WeatherIcons>
                    </div>
                    <div>
                      temp-min: 
                      9
                    </div>
                    <div>
                      temp-max: 
                      11
                    </div>
                    <div>
                      humidity: 
                      10
                    </div>
                  </div>
                </ForecastListItem>
              </div>
            </div>
          </div>
        </Forecast>
      </Connect(Forecast)>
    </Provider>
  </Router>
</MemoryRouter>
`;
